docs/js-guide.md

```diff
@@ -33,17 +36,58 @@ Projects may use any ES language features which meet all of the following criter
 As an exception, [JSX](https://reactjs.org/docs/introducing-jsx.html) is used for
 creating UI components.
 
+### Module naming
+
+Modules should use lowercase, hyphen-separated names (`helpful-module.js`).
+
+**Exception:** Modules whose primary export is a Preact component should use a
+CamelCase name that matches the component name (eg. `ShinyButton.js`). _This is a
+React/Preact ecosystem convention._
+
+### Import structure
+
+Imports should be placed at the top of a module, grouped into sections by their
+relation to the current project:
+
+- Platform (node, browser) imports
+- Third-party imports
+- Imports from other directories in the current package
+- Imports from the same directory in the current package
+
+Within each group import lines should be sorted alphabetically by module path.
```

Comment:
I have a lingering minor confusion on this: is this alphabetization by the entire path as a single string? I struggle with this, oddly! Here's a good example from a client module:

```
import { useStoreProxy } from '../store/use-store';
import { withServices } from '../service-context';
import VersionData from '../helpers/version-data';
```

Should this be:

```
import VersionData from '../helpers/version-data';
import { withServices } from '../service-context';
import { useStoreProxy } from '../store/use-store';
```

without regard to path depth?

p.s. I think I'm the only one confused by this!

Reply:
Entire path as a single string, so matching your second example. I'd hope we can automate this in future, though I haven't got a working setup yet.